core/src/components/router-outlet/route-outlet.tsx

```diff
@@ -41,7 +41,7 @@ export class RouterOutlet implements ComponentInterface, NavOutlet {
   /**
    * The mode determines which platform styles to use.
    */
-  @Prop({ mutable: true }) mode = getIonMode(this);
+  @Prop({ mutable: true }) mode = getIonPlatform(this);
```

Comment:
Does setting the mode on a `router-outlet` also affect any components rendered as part of the routing? If so, this doesn't seem correct since we'd be mixing the behaviors of `mode` and `platform`. For example, if the `platform` is globally set to `ios` and the `mode` globally set to `base`, this would result in child components using `ios` styling as well as behavior, instead of base styling.

Reply:
Yes, `mode` is inherited to child components.

`mode` controls the look and the feel of an application. `baseComponents`/`useBase` overrides the look that is set by default with `mode`. In other words, it takes priority over whatever is set using `mode` even if you set the `mode` directly on a component.

For example, `<ion-button mode="md" use-base="true">` would remove any MD styles, even though the mode is explicitly being set here. Doing it this way gives the developers control over both the look and the feel without being able to mix ineligible values. For example, the developer could never apply the iOS look to a component running with MD capabilities.

The same is true for `ion-router-outlet` here. `getIonPlatform` will default to whatever is set in the global `mode` (`'ios'` or `'md'`). However, if you have `use-base="true"` set on a child component, that component will not load ios/md stylesheets.

Reply:
Ahh, got it, I'd gotten myself turned around with how the API works. Thanks for explaining 👍